### PR TITLE
Docker Compose version 2 compatibility.

### PR DESCRIPTION
**Pull request by slr71**
Version 2 of the Docker Compose plugin has a couple of breaking changes:

1. The command is now integrated into the `docker` binary and is executed using `docker compose` rather than `docker-compose`.
2. Networks that exist already cause an error unless the network is specified in the Docker Compose file and marked as external.

To address the first issue, I updated `road-runner` to detect whether a `docker-compose` executable is present on the system. If not, it assumes that Docker Compose version 2 is being used and runs `docker compose` rather than `docker-compose`. Since `road-runner` now has to check the path for `docker-compose`, I also removed the `--docker-compose-path` and `--docker-path` command line options (which are never used anyway) and had it search the path for the `docker` executable as well.

To address the second issue, I simply removed the code to create and remove the container network. Both versions of Docker Compose handle that automatically, so it's not needed anyway.

While I was modifying the source, I switched to Go version 1.21 as well.

### CODE REVIEW

File: commands.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import (
+	"context"
+	"os/exec"
+
+	"github.com/spf13/viper"
+)
+
+// Creates a command that can be used to run docker-compose.
+func DockerComposeCommand(cfg *viper.Viper, args ...string) *exec.Cmd {
```
Comment (ianmcorvidae) [inline commands.go:11]: This is nitpicky so obviously feel free to ignore, but this is 99% duplicated with `DockerComposeCommandContext`, and `DockerCommand` is only used by this function. This function itself is only used two places. I suppose it'd only save a little duplication, but maybe it'd be better to just change those two calls to `DockerComposeCommandContext(cfg, ctx.Background(), ...)` and delete these two functions.

**Review by ianmcorvidae — APPROVED** [review]
One nitpicky thing, as seems to be my usual pattern lately, but seems fine to me, whether or not you change that one thing!

File: commands.go (hunk 1 above)
Comment (slr71) [reply commands.go:11]: That seems reasonable; I'll update that. Thanks!

Comment (slr71) [reply commands.go:11]: I went with a slightly lazier approach. I left `DockerComposeCommand` and `DockerCommand` in place but modified them to call their counterparts with `context.Background()` as the context.

**Comment by slr71** [comment]
Thanks for the review!

**Pull request merged**